src/facade/reply_builder.h

```diff
@@ -247,7 +247,8 @@ class RedisReplyBuilder : public SinkReplyBuilder {
   };
 
  private:
-  void SendStringArrInternal(WrappedStrSpan arr, CollectionType type);
+  void SendStringArrInternal(size_t size, absl::FunctionRef<std::string_view(unsigned)> producer,
+                             CollectionType type);
```

Comment:
What part of WrappedStrSpan didn't you like? The fact that we can pass it in one value or that we don't need an erased function call for that? 😆 Why didn'y you replace it everywhere?

Reply:
I kept `WrappedStrSpan` and I liked it but the interface of SendStringArrInternal is too rigid to support `SendScoredArray`. With a callback I will be able to reuse SendStringArrInternal there.